#! /usr/bin/env python

PACKAGE='yolo_classification'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                  Type       Lvl,  Description                          Def       Min     Max
gen.add('skip',               int_t,     0,    'Skip this many frames',             0,        0,      10)
gen.add('duplicate_thres',    int_t,     0,    'Threshold to check for duplicates', 20,       1,      200)
gen.add('thres',              double_t,  0,    'Darknet thres',                     0.1,      0.0,    1.0)
gen.add('hier',               double_t,  0,    'Darknet hier',                      0.0,      0.0,    1.0)
gen.add('min_prob',           double_t,  0,    'Minimum probability for output',    0.5,      0.0,    1.0)

exit(gen.generate(PACKAGE, PACKAGE, 'Darknet'))
